### DISCUSSION TITLE
Testing editing functionality of the DataTable

### DISCUSSION THREAD

**Discussion by annette-graham** [primefaces/community#1051] [PrimeReact] [answered]
Hi, I am attempting to test the functionality around editing cells using React Testing Library and Jest.

Is there any reason why these query `fireEvents` wouldn't work:
```
    fireEvent.click(getByTestId('table').querySelector('td'));
    fireEvent.change(getByTestId('table').querySelector('input'), { target: { value: 'New Value' } });
    fireEvent.blur(getByTestId('table').querySelector('input'));
```

It doesn't appear to register the input change.

Thanks!

**Comment by annette-graham** [comment]
Adding to this, is there any way a data-testid can be added to the prime react components? If not, this could be a really handy feature for testing using RTL

**Reply by melloware** [reply, marked as answer]
Yes using Passthrough: A lot of components already allow Passthrough to set the data-testid like: https://github.com/primefaces/primereact/issues/4676
